### ISSUE TITLE
Playback fails to end when reaching the end of the bar; freezes browser

### ISSUE THREAD

**Issue by j55blanchet** [j55blanchet/dance-teacher-xr-unity#120]

**Comment by j55blanchet** [comment]
Got a repro on firefox on my laptop. The following was in the console:

```
Uncaught (in promise) Error: Mismatched array lengths between matchingUserPoses, referencePoses, and frameTimes.
    calculate3DKinematicErrorDescriptors compute-kinematic-motion-descriptors-3D.ts:67
    summarizeMetric KinematicErrorMetric.ts:38
    summaryMetricResults UserDanceEvaluator.ts:207
    generateMetricSummariesForTrackSection UserDanceEvaluator.ts:205
    generatePerformanceSummary UserDanceEvaluator.ts:130
    update PracticePage.svelte:321
    update scheduler.js:115
    flush scheduler.js:79
    promise callback*schedule_update scheduler.js:20
    make_dirty Component.js:81
    ctx Component.js:139
    videowithskeleton_currentTime_binding PracticePage.svelte:1581
    videowithskeleton_currentTime_binding PracticePage.svelte:652
    ctx Component.js:138
    video_timeupdate_handler VideoWithSkeleton.svelte:333
    video_timeupdate_handler VideoWithSkeleton.svelte:66
    raf environment.js:8
    video_timeupdate_handler VideoWithSkeleton.svelte:62
    raf environment.js:8
    video_timeupdate_handler VideoWithSkeleton.svelte:62
    raf environment.js:8
    video_timeupdate_handler VideoWithSkeleton.svelte:62
    raf environment.js:8
    video_timeupdate_handler VideoWithSkeleton.svelte:62
[compute-kinematic-motion-descriptors-3D.ts:67:14](http://localhost:5173/src/lib/ai/motionmetrics/compute-kinematic-motion-descriptors-3D.ts)
```

**Comment by j55blanchet** [comment]
Two things that we should look into for addressing this issue:
1. We should be able to gracefully handle cases where metrics throw exceptions. Likely we should modify the track to support null entries for live evaluation frames, then we can wrap the metric `evaluateFrame` funciton in a try catch. We should also add a try/catch to the summary metric calculations, which can do the simple thing of putting an undefined value in the performance summary.
2. We should examine why those array lenghs were mismatched. For now, we might just improve the logging to include the actual array lengths

**Comment by j55blanchet** [comment]
I implemented logging for number 2 and got the following error:
```
Unable to summarize metric kinematicError for section bar2 Error: Mismatched array lengths between matchingUserPoses (21), referencePoses (20), and frameTimes (20).
```

**Comment by j55blanchet** [comment]
Closed as part of #128
